Examples/Sources/MacroExamples/Implementation/Member/CaseDetectionMacro.swift

```diff
@@ -22,29 +22,28 @@ public enum CaseDetectionMacro: MemberMacro {
   ) throws -> [DeclSyntax] {
     declaration.memberBlock.members
       .compactMap { $0.decl.as(EnumCaseDeclSyntax.self) }
-      .map { $0.elements.first!.name }
-      .map { ($0, $0.initialUppercased) }
-      .map { original, uppercased in
+      .map { $0.elements }
+      .flatMap { $0 }
+      .map { $0.name }
+      .map { ($0, $0.pascalcased) }
+      .map { original, pascalcased in
         """
-        var is\(raw: uppercased): Bool {
-          if case .\(raw: original) = self {
-            return true
-          }
-
-          return false
+        var is\(raw: pascalcased): Bool {
+            self == .\(raw: original)
         }
         """
       }
   }
 }
 
 extension TokenSyntax {
-  fileprivate var initialUppercased: String {
-    let name = self.text
-    guard let initial = name.first else {
-      return name
+    fileprivate var pascalcased: String {
+        self.text
+            .replacingOccurrences(of: "_", with: " ")
+            .replacingOccurrences(of: "(?<=.)([A-Z])", with: " $1", options: .regularExpression)
+            .lowercased()
+            .split(separator: " ")
```

Comment:
Do we need to insert space as the separator character here? Couldn’t we use `split(whereSeparator:)` instead? Something like

```suggestion
            .split { $0 == "_" || $0.isUppercase }
            .map { $0.lowercased().capitalized }
```

Reply:
Thanks for the suggestion! I think we could simplify this a bit further and make it more reliable for cases like enRoute or EN_ROUTE by using a regex to normalize everything into a consistent underscore-separated format first. That way, we avoid dealing with space insertion or manual splitting logic.

What do you think about something like this instead?

```
.replacingOccurrences(of: "([a-z])([A-Z])", with: "$1_$2", options: .regularExpression)
.split(separator: "_")
.map { $0.lowercased().capitalized }
```

This converts enRoute to EnRoute, EN_ROUTE to EnRoute, en_route to EnRoute, etc., without needing to match on uppercase characters manually or insert spaces. Let me know what you think!

Reply:
Actually thinking about this again, I would argue to stick with the old `initialUppercased` approach because cases that adhere to Swift’s standard naming conventions would be lowercase camel case anyway, so none of this matters. For anything else, `initialUppercased` still produces understandable results and doesn’t break cases like `v5_2`, which might be named like this to indicate a version `5.2`. Anything but `initialUppercased` would require us to implement a fully-fledged snake_case to camelCase conversion function with all the pitfalls and edge cases these come with and doing so would require quite a bit of thought that’s unrelated to the purpose of this example.